src/main/routing/shd-topology.c

```diff
@@ -122,6 +122,54 @@ static gboolean _topology_loadGraph(Topology* top, const gchar* graphPath) {
     return TRUE;
 }
 
+/** Returns FALSE if issue parsing graph, otherwise returns TRUE.
+ * If returning TRUE, whether or not the graph is complete is stored in result.
+ */
+static gboolean _topology_isComplete(Topology* top, gboolean *result) {
+    igraph_t *graph = &top->graph;
+    igraph_vs_t vs;
+    igraph_vit_t vit;
+    int ret;
+    long int vcount = igraph_vcount(graph);
+    /*
+     * Determines if a graph is complete by:
+     * - knowning how many vertexes there are
+     * - for each vertex, count the indcident edges
+     *   - if less than the number of vertexes, it isn't a complete graph
+     * - otherwise the graph is complete
+    /* vert selector. We wall all verts */
+    ret = igraph_vs_all(&vs);
+    if (ret != IGRAPH_SUCCESS) {
+        critical("igraph_vs_all returned non-success code %i", ret);
+        return FALSE;
+    }
+    ret = igraph_vit_create(graph, vs, &vit);
+    if (ret != IGRAPH_SUCCESS) {
+        critical("igraph_vit_create returned non-success code %i", ret);
+        return FALSE;
+    }
+    while (!IGRAPH_VIT_END(vit)) {
+        igraph_vector_t iedges;
+        igraph_vector_init(&iedges, 0);
```

Comment:
There is a memory leak here.

If `igraph_vector_init` returns successfully, then the vector `iedges` needs to be deallocated using `igraph_vector_destroy` after the `iedges` is no longer needed, before the while loop loops, and before this function returns. Otherwise the memory that was allocated inside of `igraph_vector_init` will be leaked.

See:
  - http://igraph.org/c/doc/ch07.html#igraph_vector_init
  - http://igraph.org/c/doc/ch07.html#igraph_vector_destroy